File: request5.js
function twoSum(nums, target) {
  // your code here
  let all = {};

  for (let i = 0; i < nums.length; i++) {
    let difference = target - nums[i];

    if (difference in all) {
      return [all[difference], i];
    }
    all[nums[i]] = i;
  }
}
let result = twoSum([2, 11, 7, 15], 9);
console.log(result); // show [0, 2] because nums[0]+nums[2] is 9

File: request5.py
def twoSum(nums, target):
# your code here
    all = {}
    for i in range(len(nums)):
        num1 = nums[i]
        num2 = target - num1
        if num2 in all:
            return [all[num2], i]
        all[num1] = i



result=twoSum([2, 11, 7, 15], 9)
print(result) # show [0, 2] because nums[0]+nums[2] is 9
